rules/kuiper.go

```diff
@@ -0,0 +1,308 @@
+	if err != nil {
+		return streams, errors.Wrap(ErrKuiperServer, err)
+	}
+	defer res.Body.Close()
+
+	err = json.NewDecoder(res.Body).Decode(&streams)
+	if err != nil {
+		return streams, errors.Wrap(ErrMalformedEntity, err)
+	}
+	return streams, nil
+}
+
+func (k *kuiper) DescribeStream(name string) (*StreamInfo, error) {
+	url := fmt.Sprintf("%s/streams/%s", k.url, name)
+	res, err := http.Get(url)
+	if err != nil {
+		return nil, errors.Wrap(ErrKuiperServer, err)
+	}
+	if res.StatusCode == http.StatusNotFound {
+		return nil, errors.Wrap(ErrNotFound, err)
+	}
+	defer res.Body.Close()
+
+	var stream StreamInfo
+	err = json.NewDecoder(res.Body).Decode(&stream)
+	if err != nil {
+		return nil, errors.Wrap(ErrMalformedEntity, err)
+	}
+
+	return &stream, nil
+}
+
+func (k *kuiper) Drop(name, kuiperType string) (*http.Response, error) {
+	if _, ok := KuiperType[kuiperType]; !ok {
+		return nil, ErrMalformedEntity
+	}
+
+	url := fmt.Sprintf("%s/%s/%s", k.url, kuiperType, name)
+	req, err := http.NewRequest(http.MethodDelete, url, nil)
+	if err != nil {
+		return nil, errors.Wrap(ErrKuiperServer, err)
+	}
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(ErrKuiperServer, err)
+	}
+	return res, nil
+}
+
+func (k *kuiper) CreateRule(rule Rule) (*http.Response, error) {
+	k.ruleURL(&rule)
+
+	body, err := json.Marshal(rule)
+	if err != nil {
+		return nil, ErrMalformedEntity
+	}
+
+	url := fmt.Sprintf("%s/rules", k.url)
+	res, err := http.Post(url, contentType, bytes.NewBuffer(body))
+	if err != nil {
+		return nil, errors.Wrap(ErrKuiperServer, err)
+	}
+	return res, nil
+}
+
+func (k *kuiper) UpdateRule(rule Rule) (*http.Response, error) {
+	k.ruleURL(&rule)
+
+	body, err := json.Marshal(rule)
+	if err != nil {
+		return nil, ErrMalformedEntity
+	}
+
+	url := fmt.Sprintf("%s/rules/%s", k.url, rule.ID)
+	req, err := http.NewRequest(http.MethodPut, url, bytes.NewBuffer(body))
+	if err != nil {
+		return nil, errors.Wrap(ErrKuiperServer, err)
+	}
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(ErrKuiperServer, err)
+	}
+
+	return res, nil
+}
+
+func (k *kuiper) ShowRules() ([]RuleInfo, error) {
+	var ruleInfos []RuleInfo
+
+	url := fmt.Sprintf("%s/rules", k.url)
+	res, err := http.Get(url)
+	if err != nil {
+		return ruleInfos, errors.Wrap(ErrKuiperServer, err)
+	}
+	defer res.Body.Close()
+
+	err = json.NewDecoder(res.Body).Decode(&ruleInfos)
+	if err != nil {
+		return ruleInfos, errors.Wrap(ErrMalformedEntity, err)
+	}
+
+	return ruleInfos, nil
+}
+
+func (k *kuiper) DescribeRule(name string) (*Rule, error) {
+	url := fmt.Sprintf("%s/rules/%s", k.url, name)
+
+	res, err := http.Get(url)
+	if err != nil {
+		return nil, errors.Wrap(ErrKuiperServer, err)
+	}
+	if res.StatusCode == http.StatusNotFound {
+		return nil, errors.Wrap(ErrNotFound, err)
+	}
+	defer res.Body.Close()
+
+	var rule Rule
+	err = json.NewDecoder(res.Body).Decode(&rule)
+	if err != nil {
```

Comment:
Because in `service.go` I just do 

```
	rule, err := re.kuiper.DescribeRule(prepend(ui.Id, name))
	if err != nil {
		return Rule{}, err
	}
```

so I want to make sure that we return an error defined in the service (and that is handled adequately in `transport.go`).